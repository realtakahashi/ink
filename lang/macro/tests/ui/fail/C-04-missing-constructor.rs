use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
