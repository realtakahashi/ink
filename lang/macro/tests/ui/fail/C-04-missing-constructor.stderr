error: ink! contracts require at least one `#[ink(constructor)]`
  --> $DIR/C-04-missing-constructor.rs:4:1
   |
4  | / mod noop {
5  | |     #[ink(storage)]
6  | |     struct Noop {}
7  | |
...  |
11 | |     }
12 | | }
   | |_^
